src/ameba/source.cr

```diff
@@ -57,6 +57,22 @@ module Ameba
     # source.ast
     # ```
     getter ast : Crystal::ASTNode do
+      code = @code
+
+      if Ameba.ecr_supported? && @path.ends_with?(".ecr")
+        begin
+          code = ECR.process_string(code, @path)
+        rescue ex : ECR::Lexer::SyntaxException
```

Comment:
Looks like there's a `require` missing, covering that path.

Reply:
There is not, the problem is alpine is still using Crystal 1.14, which this requires 1.15

Reply:
Missing `require` here is not a problem indeed, yet the reason that builds fail on Crystal 1.14 is due to the fact, that the check for ecr support moved from compile-time to runtime, which breaks builds on Crystal < 1.15, that's why the `ecr_supported?` method needs to turned into a macro.